epoxy-viewbinder/src/main/java/com/airbnb/epoxy/EpoxyViewBinder.kt

```diff
@@ -0,0 +1,480 @@
+ * APK and the dynamic feature.
+ * @param modelProvider a lambda for building the [EpoxyModel]. It expects a single model to be
+ * added to the controller receiver. If no model is added the view will be hidden.
+ */
+class LifecycleAwareEpoxyViewBinder(
+    private val lifecycleOwner: LifecycleOwner,
+    private val rootView: (() -> View?),
+    @IdRes private val viewId: Int,
+    private val useVisibilityTracking: Boolean = false,
+    private val fallbackToNameLookup: Boolean = false,
+    private val modelProvider: EpoxyController.() -> Unit,
+) : LifecycleObserver {
+    private val viewBinder = EpoxyViewBinder()
+    private var lazyView: View? = null
+
+    private val visibilityTracker: EpoxyViewBinderVisibilityTracker by lazy {
+        EpoxyViewBinderVisibilityTracker().apply {
+            this.partialImpressionThresholdPercentage = 100
+        }
+    }
+
+    val view: View
+        get() {
+            if (lazyView == null) {
+                val nonNullRootView = rootView() ?: error("Fragment view is not created")
+                lazyView = nonNullRootView.maybeFindViewByIdName(viewId, fallbackToNameLookup)
+                    ?: error(
+                        "View could not be found, fallbackToNameLookup: $fallbackToNameLookup," +
+                            " view id name: ${nonNullRootView.resources.getResourceEntryName(viewId)}"
+                    )
+                // Propagate an error if a non EpoxyViewStub is used
+                if (lazyView !is EpoxyViewStub) {
+                    val resourceNameWithFallback = try {
+                        nonNullRootView.resources.getResourceName(viewId)
+                    } catch (e: Resources.NotFoundException) {
+                        "$viewId (name not found)"
+                    }
+                    viewBinder.onExceptionSwallowed(
+                        IllegalStateException(
+                            "View binder should be using EpoxyViewStub. View ID: $resourceNameWithFallback"
+                        )
+                    )
+                }
+
+                // Register this for view lifecycle callbacks so that it can clear the view when it
+                // is destroyed. This both prevents a memory leak, and ensures that if the view is
+                // recreated it can look up the reference again. This MUST register the observer
+                // again each time the view is created because the fragment's viewLifecycleOwner
+                // is updated to a new instance for each new fragment view.
+                lifecycleOwner.lifecycle.addObserver(this)
+            }
+
+            return lazyView!!
+        }
+
+    /** Adds an [EpoxyController.Interceptor] to the [EpoxyViewBinder]. **/
+    fun addInterceptor(interceptor: EpoxyController.Interceptor) {
+        viewBinder.addInterceptor(interceptor)
+    }
+
+    /** Replaces the [View] with the model produced by the [modelProvider] lambda. */
```

Comment:
This is probably too simplistic of a kdoc for this function as it could either hide, replace, or update the view depending on the prior and new model. I'll update it.